src/pages/clipboard.astro

```diff
@@ -0,0 +1,283 @@
+                <span class="p-2" data-clipboard-target="ctaText">Copy</span>
+                <span
+                  class="opacity-0 invisible duration-200 rounded translate-y-2 absolute inset-0 flex items-center justify-center"
+                  data-clipboard-target="successText"
+                >
+                  Copied!
+                </span>
+              </Button>
+            </div>
+          </div>
+        </div>
+
+        <div slot="code">
+          <Markdown>
+            ```html
+            <div
+              data-controller="clipboard"
+              data-clipboard-toggle-success-class="opacity-0 invisible translate-y-2"
+            >
+              <input
+                type="text"
+                value="Input text to be copied"
+                data-clipboard-target="source"
+              >
+              <button
+                class="relative overflow-hidden"
+                data-action="clipboard#copy"
+              >
+                <span data-clipboard-target="ctaText">Copy</span>
+                <span
+                  class="opacity-0 invisible duration-200 translate-y-2 absolute inset-0 flex items-center justify-center"
+                  data-clipboard-target="successText"
+                >
+                  Copied!
+                </span>
+              </button>
+            </div>
+            ```
+          </Markdown>
+        </div>
+      </CodeDemo>
+      
+      <CodeDemo heading="Custom success message with dispatched events" className="py-8">
+        <div slot="preview">
+          <div
+            class="max-w-md mx-auto"
+            data-controller="clipboard"
+            data-clipboard-success-delay-value="1500"
+          >
+            <input
+              type="hidden"
+              value={Astro.request.url}
+              data-clipboard-target="source"
+            >
+            <div class="flex justify-center">
+              <Button className="relative overflow-hidden" data-action="clipboard#copy">
+                Copy URL
+              </Button>
+            </div>
+          </div>
+          <div
+            data-controller="class-list"
+            data-class-list-change-class="opacity-0 translate-y-[120%] invisible"
+            data-class-list-use-classes-value="true"
+            data-action="clipboard:copy@document->class-list#remove clipboard:success-dismissed@document->class-list#add"
+          >
+            <div
+              class="opacity-0 translate-y-[120%] invisible duration-200 absolute bottom-2 right-2 rounded bg-white bg-opacity-20 py-2 px-4"
+              data-class-list-target="receiver"
+            >
+              Copied to clipboard
+            </div>
+          </div>
+        </div>
+
+        <div slot="code">
+          <Markdown>
+            ```html
+            <div
+              data-controller="clipboard"
+              data-clipboard-success-delay-value="1500"
+            >
+              <input
```

Comment:
Seeing this hidden input makes me wonder if we could also have an optional Stiumulus value that we can copy from. 

Just so the following usecase doesn't need the extra boilerplate of a hidden input.

Reply:
🤦 such a good point, will add